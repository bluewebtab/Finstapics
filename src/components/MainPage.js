import  React, { Component } from 'react';
import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import ErrorBoundry from '../components/ErrorBoundry';
import Header from '../components/Header';


   
   
    class MainPage extends Component {

      

      componentDidMount() {
       this.props.onRequestImages();
       
      }

      filterImages = () => {
         //filter function is used filter the names in order to find an item
         return this.props.images.filter(image => {
          return image.user.first_name.toLowerCase().includes(this.props.searchField.toLowerCase())
        })
      }


      render(){
        const { onSearchChange, isPending} = this.props;

        return (
            <div>
              <Header />
              <SearchBox searchChange={onSearchChange}/>
              <Scroll>
           
                {isPending ? <h1>Loading</h1> : 
                    <ErrorBoundry>
                      <CardList data={this.filterImages()}/>
                    </ErrorBoundry>
                }
              </Scroll>
              
              </div>
              
         
         
         
        
             )
      
      }

    }

    export default MainPage;