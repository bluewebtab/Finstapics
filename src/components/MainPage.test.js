import {shallow, mount, render} from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

let wrapper;

beforeEach(() => {
   const mockProps = {
    onRequestImages: jest.fn(),
    images: [],
    searchField: '',
    isPending: false
   }
   wrapper = shallow(<MainPage {...mockProps}/>)
})

it('renders MainPage without crashing', () => {
    expect(wrapper).toMatchSnapshot();
})



it('filters images correctly', () => {
    const mockProps2 = {
        onRequestImages: jest.fn(),
        images: [{
            id: 1,
            urls: {
                    small: "https://images.unsplash.com/photo-1587613757703-eea60bd69e66?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&ixid=eyJhcHBfaWQiOjEzMTk1Mn0"
            },
            user: {
                first_name: "Hello"
            },
            likes: 3
        }],
        searchField: 'Hello',
        isPending: false
       }
      const wrapper2 = shallow(<MainPage {...mockProps2} />)
    //expect filterImages to run an empty array
    expect(wrapper2.instance().filterImages()).toEqual([{
        id: 1,
        urls: {
                small: "https://images.unsplash.com/photo-1587613757703-eea60bd69e66?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&ixid=eyJhcHBfaWQiOjEzMTk1Mn0"
        },
        user: {
            first_name: "Hello"
        },
        likes: 3
    }])

})

it('filters images correctly 2', () => {
    const mockProps3 = {
        onRequestImages: jest.fn(),
        images: [{
            id: 1,
            urls: {
                    small: "https://images.unsplash.com/photo-1587613757703-eea60bd69e66?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=400&fit=max&ixid=eyJhcHBfaWQiOjEzMTk1Mn0"
            },
            user: {
                first_name: "Hello"
            },
            likes: 3
        }],
        searchField: 'a',
        isPending: false
       }
       const filteredImages = []
      const wrapper3 = shallow(<MainPage {...mockProps3} />)
    //expect filterImages to run an empty array
    expect(wrapper3.instance().filterImages()).toEqual(filteredImages)

})


 
