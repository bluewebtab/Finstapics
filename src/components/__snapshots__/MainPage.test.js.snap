// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks if a function is passed as prop 1`] = `ShallowWrapper {}`;

exports[`re 1`] = `ShallowWrapper {}`;

exports[`renders MainPage without crashing 1`] = `ShallowWrapper {}`;

exports[`renders SearchBox without crashing 1`] = `ShallowWrapper {}`;
